File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02718/p02718_num6_parsed.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int main() {
    int N, M;
    std::cin >> N >> M;

    std::vector<int> votes(N);
    for (int i = 0; i < N; ++i) {
        std::cin >> votes[i];
    }

    std::sort(votes.rbegin(), votes.rend());

    int totalVotes = 0;
    for (int i = 0; i < M; ++i) {
        totalVotes += votes[i];
    }

    int threshold = totalVotes / (4 * M);

    if (votes[M-1] >= threshold) {
        std::cout << "Yes";
    } else {
        std::cout << "No";
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02718/p02718_num6_parsed.py
import sys

def can_select_popular_items(N, M, votes):
    total_votes = sum(votes)
    threshold = total_votes / (4 * M)
    popular_items = [vote for vote in votes if vote >= threshold]
    
    if len(popular_items) >= M:
        return 'Yes'
    else:
        return 'No'

# read input from STDIN
N, M = map(int, input().split())
votes = list(map(int, input().split()))

# call function and print the result
result = can_select_popular_items(N, M, votes)
print(result)